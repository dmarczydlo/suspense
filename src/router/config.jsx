import React, {lazy} from 'react';
import {Route} from 'react-router-dom';

import Home from '../sites/home';

export const routes = [
    {
        component: Home,
        exact: true,
        path: '/'
    },
    {
        component: Home,
        path: '/child/:id'
    }
];

export const RouteWithSubRoutes = (route, key) => (
    <Route
        exact={route.exact}
        key={key}
        path={route.path}
        render={(props) => (
            <route.component
                {...props}
                routes={route.routes}
            />
        )}
    />
);
