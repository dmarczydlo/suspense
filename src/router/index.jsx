import React from 'react';
import {BrowserRouter as Router, Route, Switch} from 'react-router-dom';
import {RouteWithSubRoutes, routes} from './config';
import NotFound from '../sites/notFound';
import Layout from '../layout';

const RouterElements = () => {
    return (
        <Router>
            <Layout>
                <Switch>
                    {routes.map((route) => (
                        <RouteWithSubRoutes
                            {...route}
                            key={route.path}
                        />
                    ))}
                    <Route component={NotFound} />
                </Switch>
            </Layout>
        </Router>
    );
};

export default RouterElements;
