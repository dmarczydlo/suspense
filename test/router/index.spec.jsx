import React from 'react';
import {BrowserRouter as Router} from 'react-router-dom';
import Index from '../../src/router/index';
import {routes, RouteWithSubRoutes as routeWithSubRoutes} from '../../src/router/config';

describe('Router index', () => {
    let wrapper = null;

    it('should be rendered', () => {
        wrapper = global.shallow(<Index />);
        expect(wrapper).toMatchSnapshot();
    });

    it('Config should be rendered', () => {
        const {0: route} = routes;
        expect(routeWithSubRoutes(route)).not.toBeNull();
        wrapper = global.shallow(
            <Router>
                {routeWithSubRoutes(route)}
            </Router>);
        expect(wrapper).toMatchSnapshot();
    });

    it('Config should be rendered without layout', () => {
        const Test = () => (
            <span>
                {'test'}
            </span>
        );
        const route = {
            component: Test,
            layout: false,
            path: '/test/'
        };
        expect(routeWithSubRoutes(route)).not.toBeNull();
        wrapper = global.shallow(
            <Router>
                {routeWithSubRoutes(route)}
            </Router>
        );
        expect(wrapper).toMatchSnapshot();
    });
});
