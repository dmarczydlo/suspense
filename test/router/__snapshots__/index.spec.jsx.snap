// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router index Config should be rendered 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
</Router>
`;

exports[`Router index Config should be rendered without layout 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Route
    path="/test/"
    render={[Function]}
  />
</Router>
`;

exports[`Router index should be rendered 1`] = `
<BrowserRouter>
  <Index>
    <Switch>
      <RouteWithSubRoutes
        component={[Function]}
        exact={true}
        key="/"
        path="/"
      />
      <RouteWithSubRoutes
        component={[Function]}
        key="/child/:id"
        path="/child/:id"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </Index>
</BrowserRouter>
`;
